pkg/planner/context/context.go

```diff
@@ -72,3 +73,20 @@ type EmptyPlanContextExtended struct{}
 
 // AdviseTxnWarmup advises the txn to warm up.
 func (EmptyPlanContextExtended) AdviseTxnWarmup() error { return nil }
+
+// GetRangerCtx creates a RangerContext from PlanContext
+func GetRangerCtx(sctx PlanContext) *rangerctx.RangerContext {
```

Comment:
We use `GetRangerCtx` instead of giving a method `PlanContext.GetRangerCtx` because dependency cycle?

Reply:
No. Maybe it'd be better to add a method `PlanContext.GetRangerCtx`. I used `GetRangerCtx` because it only depends on the public interface defined in `PlanContext`, but now I think it'd be better to add a method because maybe `GetSessionVars()` (etc) will be removed finally 🤔 .

I'll do it right now.

Reply:
I think I can add this method in `PlanCtxExtendedImpl`. It looks cool 🤔

Reply:
Done. PTAL

Reply:
Done. PTAL